libs/cart/src/drivers/magento/cart-coupon.service.spec.ts

```diff
@@ -0,0 +1,390 @@
+    describe('when the call to the Magento API is successful', () => {
+      beforeEach(() => {
+        mockRemoveAllCouponsResponse.removeCouponFromCart.cart.applied_coupons = [];
+      });
+
+      it('should return a cart without the coupon applied', done => {
+        service.remove(cartId, mockDaffCartCoupon).subscribe(result => {
+          expect(result.coupons).not.toContain(mockDaffCartCoupon);
+          done();
+        });
+
+        const op = controller.expectOne(removeAllCoupons);
+
+        op.flush({
+          data: mockRemoveAllCouponsResponse
+        });
+      });
+    });
+
+    describe('when the call to the Magento API is unsuccessful', () => {
+      describe('because of a graphql-no-such-entity error', () => {
+        it('should throw a DaffCartNotFoundError', done => {
+          service.remove(cartId, mockDaffCartCoupon).pipe(
+            catchError(err => {
+              expect(err).toEqual(jasmine.any(DaffCartNotFoundError));
+              done();
+              return [];
+            })
+          ).subscribe();
+
+          const op = controller.expectOne(removeAllCoupons);
+
+          op.graphqlErrors([new GraphQLError(
+            'Can\'t find a cart with that ID.',
+            null,
+            null,
+            null,
+            null,
+            null,
+            {category: 'graphql-no-such-entity'}
+          )]);
+        });
+      });
+
+      describe('because of a graphql-input error', () => {
+        it('should throw a DaffBadInputError', done => {
+          service.remove(cartId, mockDaffCartCoupon).pipe(
+            catchError(err => {
+              expect(err).toEqual(jasmine.any(DaffBadInputError));
+              done();
+              return [];
+            })
+          ).subscribe();
+
+          const op = controller.expectOne(removeAllCoupons);
+
+          op.graphqlErrors([new GraphQLError(
+            'The required coupon_code argument contains an empty value.',
+            null,
+            null,
+            null,
+            null,
+            null,
+            {category: 'graphql-input'}
+          )]);
+        });
+      });
+    });
+
+    afterEach(() => {
+      controller.verify();
+    });
+  });
+
+  describe('removeAll | removing all coupons from the specified cart', () => {
+    describe('when the call to the Magento API is successful', () => {
+      beforeEach(() => {
+        mockRemoveAllCouponsResponse.removeCouponFromCart.cart.applied_coupons = [];
+      });
+
+      it('should return a cart with no coupons', done => {
+        service.removeAll(cartId).subscribe(result => {
+          expect(result.coupons).toEqual([]);
+          done();
+        });
+
+        const op = controller.expectOne(removeAllCoupons);
+
+        op.flush({
+          data: mockRemoveAllCouponsResponse
+        });
+      });
+    });
+
+    describe('when the call to the Magento API is unsuccessful', () => {
```

Comment:
all of the groups of two tests for error can be just one test, because the transforming of the error should be tested in the transformCartMagentoError. Can just have one, "when the call to the Magento API fails, should throw an error".

Reply:
actually, I guess I could kind of go either way with this. It's fine as is